internal/clientconn/conn.go

```diff
@@ -387,28 +388,39 @@ func (c *conn) route(connCtx context.Context, reqHeader *wire.MsgHeader, reqBody
 		connCtx, span = otel.Tracer("").Start(connCtx, "")
 
 		if err == nil {
-			resMsg := c.handleOpMsg(connCtx, &middleware.MsgRequest{OpMsg: msg}, command)
-			resBody = resMsg.OpMsg
+			var cmdHandler middleware.MsgHandlerFunc
+
+			cmd, ok := c.h.Commands()[command]
+			if ok && cmd.Handler != nil {
+				cmdHandler = cmd.Handler
+			} else {
+				cmdHandler = notFound(command)
+			}
+
+			var res *middleware.MsgResponse
+			res, err = cmdHandler(connCtx, &middleware.MsgRequest{OpMsg: msg})
```

Comment:
Handling error here fixes metrics and traces to record errors, which wasn't previously the case.